modules/puppet_proxy_puppet_api/v3_environment_classes_api_classes_retriever.rb

```diff
@@ -96,7 +96,10 @@ def async_get_classes(environment)
         end
       end
     end
-    @m.synchronize { @futures_cache[environment] = future }
+    @m.synchronize do
+      @futures_cache[environment] = future
+      logger.debug { "Class cache: #{@classes_cache.size}, ETag cache: #{@etag_cache.size}, Futures cache: #{@futures_cache.size}" }
```

Comment:
It doesn't seem the logger has an implicit prefix, so it would be good to explicitly mention where this lines are coming from: something like `"Puppet cache counts: classes %d, ETag %d,  futures %d"`

Reply:
Also, I wonder, why we're only logging it at this specific place? (what about other ones where we touch any of the cache variables?)

Reply:
Not sure what you mean by the prefix, but rebased as you suggested.